core-blocks/audio/edit.js

```diff
@@ -36,6 +38,30 @@ class AudioEdit extends Component {
 		this.onSelectURL = this.onSelectURL.bind( this );
 	}
 
+	componentDidMount() {
+		const { attributes, noticeOperations, setAttributes } = this.props;
+		const { id, src = '' } = attributes;
+
+		if ( ! id && src.indexOf( 'blob:' ) === 0 ) {
+			const file = getBlobByURL( src );
+
+			if ( file ) {
+				editorMediaUpload( {
+					filesList: [ file ],
+					onFileChange: ( [ { url } ] ) => {
+						setAttributes( { src: url } );
+					},
+					onError: ( e ) => {
+						setAttributes( { src: undefined, id: undefined } );
+						this.setState( { editing: true } );
+						noticeOperations.createErrorNotice( e );
```

Comment:
I like the mechanism used to display error notices which are not displayed in other cases. We should probably update existing code to also offer this possibility.